### PR TITLE
Fix broken TOC of PDFs if section includes an image

### PR DESCRIPTION
**Pull request by misery**
If someone adds an image to a section it will be added to the sidebar
TOC of a PDF, too. A PDF viewer will show the "width and height"
data instead of an image. So we need to filter out images for this with
the optional "short title" of latex sections.

### CODE REVIEW

**Comment by misery** [comment]
See https://github.com/sphinx-doc/sphinx/pull/2045

**Comment by misery** [comment]
Is there anything I can do?

**Comment by lehmannro** [comment]
What's the difference to #2045?

**Comment by misery** [comment]
Just a rebase from master to stable and changes to current stable to avoid merge conflicts.

File: tests/test_build_latex.py [hunk 1]
```diff
@@ -423,6 +423,19 @@ def test_latex_show_urls_is_no(app, status, warning):
             '{sphinx-dev@googlegroups.com}\n' in result)
 
 
+@with_app(buildername='latex', testroot='image')
```
Comment (tk0miya) [inline tests/test_build_latex.py]: I feel `test-image` is ambiguous.
Could you rename `test-image`?

**Comment by tk0miya** [comment]
LGTM.
I commented on the naming of test files.
Could you rename it please?

@lehmannro Do you still have any opinions or comments?
If none, I'll merge this soon.

(I'll update CHANGES on merging. it's too fragile to put in PR)

**Comment by lehmannro** [comment]
LGTM.

File: tests/test_build_latex.py (hunk 1 above)
Comment (misery) [inline tests/test_build_latex.py]: Done!

**Comment by tk0miya** [comment]
Merged! Thank you for contribution.

@lehmannro Thank you for review and confirmation.

**Pull request merged**